_examples/ingress/main.tf

```diff
@@ -158,6 +158,58 @@ resource "kubernetes_service" "cheddar" {
   }
 }
 
+resource "kubernetes_deployment" "stilton" {
+  metadata {
+    name = "stilton-cheese"
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "stilton"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "stilton"
+        }
+      }
+
+      spec {
+        container {
+          name  = "stilton"
+          image = "errm/cheese:stilton"
+
+          port {
+            container_port = 80
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "kubernetes_service" "stilton" {
+  metadata {
+    name = "stilton"
+  }
+
+  spec {
+    selector = {
+      app = "stilton"
+    }
+
+    port {
+      port        = 80
+      target_port = 80
+    }
+
+    type = "NodePort"
+  }
+}
+
 output "ingress_ip" {
-  value = formatlist("%s ", kubernetes_ingress.example.load_balancer_ingress.*.ip)
```

Comment:
I'm not sure it's a good idea to drop the `formatlist` here. The reason it's there is because if the list of IPs is empty, that output value will actually throw an error.

Reply:
I faced such an error when I used `[0]` instead of the splat operator, let me recheck from scratch.

Reply:
Works fine:

```
Apply complete! Resources: 7 added, 0 changed, 0 destroyed.

Outputs:

ingress_ip = []
```

Reply:
Then after a while:

```
# terraform apply -auto-approve
kubernetes_ingress.example: Refreshing state... [id=default/example]
kubernetes_service.echoserver: Refreshing state... [id=default/echoserver]
kubernetes_service.cheddar: Refreshing state... [id=default/cheddar]
kubernetes_service.stilton: Refreshing state... [id=default/stilton]
kubernetes_deployment.stilton: Refreshing state... [id=default/stilton-cheese]
kubernetes_deployment.echoserver: Refreshing state... [id=default/echoserver]
kubernetes_deployment.cheddar: Refreshing state... [id=default/cheddar-cheese]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

ingress_ip = []

# terraform apply -auto-approve
kubernetes_service.stilton: Refreshing state... [id=default/stilton]
kubernetes_ingress.example: Refreshing state... [id=default/example]
kubernetes_service.echoserver: Refreshing state... [id=default/echoserver]
kubernetes_service.cheddar: Refreshing state... [id=default/cheddar]
kubernetes_deployment.echoserver: Refreshing state... [id=default/echoserver]
kubernetes_deployment.cheddar: Refreshing state... [id=default/cheddar-cheese]
kubernetes_deployment.stilton: Refreshing state... [id=default/stilton-cheese]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

ingress_ip = [
  "34.96.95.54",
]
```

Reply:
Must have been something that got fixed in 1.12.1
I was testing those changes with 1.12.0 right after release.
I'll try to reproduce and confirm tomorrow.

Reply:
I've reproduced the issue with minikube as it failed to allocate an IP address:

```
# terraform apply
...
kubernetes_service.example: Still creating... [9m40s elapsed]
kubernetes_service.example: Still creating... [9m50s elapsed]
kubernetes_service.example: Still creating... [10m0s elapsed]

Error: Waiting for service "default/terraform-nginx-example" to assign IP/hostname for a load balancer

  on service.tf line 1, in resource "kubernetes_service" "example":
   1: resource "kubernetes_service" "example" {


# terraform apply
kubernetes_replication_controller.example: Refreshing state... [id=default/terraform-nginx-example]
kubernetes_service.example: Refreshing state... [id=default/terraform-nginx-example]

Error: Invalid index

  on service.tf line 20, in output "lb_ip":
  20:   value = kubernetes_service.example.load_balancer_ingress[0].ip
    |----------------
    | kubernetes_service.example.load_balancer_ingress is empty list of object

The given key does not identify an element in this collection value.

```